server/datastore/mysql/software_installers.go

```diff
@@ -1124,16 +1166,42 @@ WHERE
 }
 
 func (ds *Datastore) vppAppJoin(appID fleet.VPPAppID, status fleet.SoftwareInstallerStatus) (string, []interface{}, error) {
+	// for pending status, we'll join through upcoming_activities
+	if status == fleet.SoftwarePending || status == fleet.SoftwareInstallPending || status == fleet.SoftwareUninstallPending {
+		stmt := `JOIN (
+SELECT DISTINCT
+	host_id
+FROM 
+	upcoming_activities ua 
+	JOIN vpp_app_upcoming_activities vppua ON ua.id = vppua.upcoming_activity_id
+WHERE 
+	%s) hss ON hss.host_id = h.id`
+
+		filter := "vppua.adam_id = ? AND vppua.platform = ?"
+		switch status {
+		case fleet.SoftwareInstallPending:
+			filter += " AND ua.activity_type = 'vpp_app_install'"
+		case fleet.SoftwareUninstallPending:
+			// VPP does not have uninstaller yet so to preserve existing behavior of VPP filters we map uninstall to install
```

Comment:
It seems the old behaviour had `SoftwarePending` mapped to `SoftwareInstallPending`, doesn't look like it supported uninstall?